python/reduction.cpp

```diff
@@ -61,6 +61,30 @@ template <class T> void bind_sum_out(py::module &m) {
       py::call_guard<py::gil_scoped_release>());
 }
 
+template <class T> void bind_nansum(py::module &m) {
+  if constexpr (std::is_same_v<T, Variable>) {
+    m.def(
+        "nansum",
+        [](const typename T::const_view_type &x) { return nansum(x); },
+        py::arg("x"), py::call_guard<py::gil_scoped_release>());
+  }
```

Comment:
Since you are touching the related files anyway: Maybe this is a good opportunity to also provide this overload for `DataArray`, and same for `sum` above?